staging/src/k8s.io/apimachinery/pkg/semanticrv/comparison.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package semanticrv
+
+// Comparison is the result of comparing two values in a partial order.
+type Comparison struct {
+	LE bool // less than or equal
+	GE bool // greater than or equal
+}
+
+func Equal() Comparison { return Comparison{LE: true, GE: true} }
+
+func Less() Comparison { return Comparison{LE: true, GE: false} }
+
+func Greater() Comparison { return Comparison{LE: false, GE: true} }
```

Comment:
we explicitly don't want to expose Less/Greater comparisons for RVs, right?

Reply:
I am told that there is a lot of code that parses ResourceVersion as integer, and I can only assume that most of that is because said code is not satisfied with simply comparing for equality.

Reply:
There's some, not a lot, but we explicitly don't want clients parsing resourceVersion to do any comparison other than equality:

> Resource versions are strings that identify the server's internal version of an object. Resource versions can be used by clients to determine when objects have changed, or to express data consistency requirements when getting, listing and watching resources. Resource versions must be treated as opaque by clients and passed unmodified back to the server.
> 
> You must not assume resource versions are numeric or collatable. API clients may only compare two resource versions for equality (this means that you must not compare resource versions for greater-than or less-than relationships).

The fact that some clients do parsing/comparisons today is unfortunate, but enshrining it in a library we provide blesses the practice.